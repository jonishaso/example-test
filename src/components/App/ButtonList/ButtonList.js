import React from 'react'
import styled from 'styled-components'

const StyledBtn = styled.button`
  background-color: #64b5f6;
  border: 0;
  border-radius: 4px;
  color: #424242;
  font-family: 'Work Sans', sans-serif;
  font-size: 18px;
  margin: 0 5px;
  padding: 7px 14px;
  width: 80px;
  text-align: center;
  &:hover {
    background-color: #bbdefb;
  }
`
const ButtonList = ({ btnValue, handleClick }) => {
  const items = btnValue?.map((value, index) => (
    <StyledBtn key={index} onClick={() => handleClick(value)}>
      {value > 0 ? `+${value}` : value}
    </StyledBtn>
  ))
  return <div id="button-list">{items}</div>
}

export default ButtonList
