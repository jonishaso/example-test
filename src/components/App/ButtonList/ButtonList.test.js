import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ButtonList from './index'
import { ColorContext } from '../ButtonThemeContext'

const mockData = {
  btnThemeColor: '#03619cG',
  setBtnThemeColor: jest.fn(),
}

test('test Category Buttons', () => {
  const wrapper = shallow(
    <ColorContext.Provider value={{ ...mockData }}>
      <ButtonList btnValue={[-23, 45, 89, 1]} handleClick={() => {}} />
    </ColorContext.Provider>,
  )
  expect(toJson(wrapper)).toMatchSnapshot()
})
