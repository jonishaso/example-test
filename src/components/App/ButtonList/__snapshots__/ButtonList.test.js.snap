// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Category Buttons 1`] = `
<ButtonList
  btnValue={
    Array [
      -23,
      45,
      89,
      1,
    ]
  }
  handleClick={[Function]}
/>
`;
